core/src/Streamly/Internal/Data/Unfold.hs

```diff
@@ -585,6 +588,8 @@ fromStreamK = Unfold step pure
             Just (x, xs) -> Yield x xs
             Nothing -> Stop) <$> K.uncons stream
 
+fromStream :: Applicative m => Unfold m (Stream.Stream m a) a
+fromStream = lmap Stream.toStreamD fromStreamD
```

Comment:
Should be implemented using fromStreamK. See the comments in the Streamly.Data.Unfold module.